Changes needed to build without errors/warnings using JDK11

starksm64 opened the pull request:
These are the changes I needed to do in order to build the project using JDK11. This still is producing JDK8 source/binary artifacts.

pzygielo commented on spec/pom.xml:
```diff
@@ -59,8 +59,8 @@
-                                    <version>[1.8.0,1.9.0)</version>
-                                    <message>You need JDK8 or lower</message>
+                                    <version>[11.0,17.0)</version>
+                                    <message>You need JDK11 or lower</message>
```
`lower` doesn't sound right here.

starksm64 replied:
Right, I have corrected that

starksm64 commented:
Note I added the module-info and updated cdi, interceptors to service releases that include a proper module-info. 

We should try to get an RC release out based on this.

tomjenkinson commented on spec/pom.xml:
```diff
@@ -59,8 +59,8 @@
                                 <requireJavaVersion>
-                                    <version>[1.8.0,1.9.0)</version>
-                                    <message>You need JDK8 or lower</message>
+                                    <version>[11.0,17.0)</version>
```
Is it correct to list `17.0` here?

starksm64 replied:
That is the upper bound we are targeting in EE10, but really it does not matter. It could be fixed to just 11. I'll leave it up to you.

starksm64 replied:
It is captured in this platform issue: https://github.com/eclipse-ee4j/jakartaee-platform/issues/331
We will come up with a more formal statement in the platform spec requirements in the future.

pzygielo replied:
Why 17+ can't be used to build this project? (`)` -> `17.0 not included` in the range)
17 is already released LTS if I'm not mistaken.
Why limit the JDK usable for building from the top at all, while `--release` flag is used?

pzygielo replied:
Only now I noticed this is change in `spec` project, and nothing is enforced in `api`. And `spec` is not included in top level project anyway.

tomjenkinson replied:
I guess it is historic to be located in this part of the pom but upon deeper reflection I think the upper bound should be removed (and if it is not the message below would not be accurate anyway. Scott, please can you remove the upper bound?

starksm64 replied:
Change to simply 11

tomjenkinson commented:
retest this please

tomjenkinson commented:
I am trying to retest with https://download.eclipse.org/jakartaee/transactions/2.0/jakarta-transactions-tck-2.0.1.zip

tomjenkinson commented:
retest this please

tomjenkinson commented:
I am trying to retest with 2.0.2 of the TCK following the comment from Alwin (https://github.com/eclipse-ee4j/jakartaee-tck/issues/812#issuecomment-1025712502)

tomjenkinson commented:
retest this please

tomjenkinson commented:
retest this please

tomjenkinson commented:
retest this please

tomjenkinson reviewed (approved):
I will mention that the current JDK11 build of `master` works without these changes:
https://ci.eclipse.org/jta/job/eclipse-ee4j_jta-api-openjdk-jdk11-latest/22/

Thanks for the contribution, I am approving.
